pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -430,14 +430,6 @@ func (c *Reconciler) createBuildPod(ctx context.Context, tr *v1alpha1.TaskRun, t
 		return nil, fmt.Errorf("translating Build to Pod: %v", err)
 	}
 
-	// Rewrite the pod's OwnerRef to point the TaskRun, instead of a
-	// non-existent build.
-	// TODO(jasonhall): Just set this directly when creating a Pod from a
-	// TaskRun.
-	pod.OwnerReferences = []metav1.OwnerReference{
```

Comment:
I moved this into `makePod`, but I'm not sure if that's ok or if we are trying to keep the code in sync with upstream for now.

Reply:
So, the MakePod function is not used at all right now. We still rely on the knative/build API to create a Build instead of using the MakePod above.

Reply:
Hmm, maybe I am misunderstanding something. As far as I can tell, we [import `pod.go` from build-pipeline](https://github.com/knative/build-pipeline/blob/709bc8516d6dd546318f35c52346e45cdecaaf79/pkg/reconciler/v1alpha1/taskrun/taskrun.go#L32) and then use that version [here](https://github.com/knative/build-pipeline/blob/709bc8516d6dd546318f35c52346e45cdecaaf79/pkg/reconciler/v1alpha1/taskrun/taskrun.go#L428) so that as of #326 we aren't actually using Knative/build, and in my local testing the changes I made to `MakePod` did affect the output. Is there some kind of flag enabling different behavior in testing vs production usage or something?

Reply:
@tejal29 : @ImJasonH submitted PR to remove dependency on Build API. Currently there is code dependency but TaskRun is not creating Build CRD object anymore. 

@dwnusbaum : There is no need for maintaining code similarity with Build code base. If it makes sense to pull the logic into separate function then please feel free to do so.